code/modules/capture_the_flag/ctf_game.dm

```diff
@@ -3,11 +3,173 @@
 #define BLUE_TEAM "Blue"
 #define GREEN_TEAM "Green"
 #define YELLOW_TEAM "Yellow"
-#define FLAG_RETURN_TIME 200 // 20 seconds
-#define INSTAGIB_RESPAWN 50 //5 seconds
-#define DEFAULT_RESPAWN 150 //15 seconds
+#define FLAG_RETURN_TIME 20 SECONDS
 
-/obj/item/ctf
+///Base CTF machines, if spawned in creates a CTF game with the provided game_id unless one already exists. If one exists associates itself with it.
+/obj/machinery/ctf
+	name = "CTF Controller"
+	desc = "Used for running friendly games of capture the flag."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "syndbeacon"
+	density = TRUE
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	///The game ID that this CTF machine is associated with.
+	var/game_id = CTF_GHOST_CTF_GAME_ID
+	///Reference to the CTF controller that this machine operates under.
+	var/datum/ctf_controller/ctf_game
+
+/obj/machinery/ctf/Initialize(mapload)
+	. = ..()
+	ctf_game = GLOB.ctf_games[game_id]
+	if(isnull(ctf_game))
+		ctf_game = create_ctf_game(game_id)
+
+///A spawn point for CTF, ghosts can interact with this to vote for CTF or spawn in if a game is running.
+/obj/machinery/ctf/spawner
+	///The team that this spawner is associated with.
+	var/team = WHITE_TEAM
+	///The span applied to messages associated with this team.
+	var/team_span = ""
+	///assoc list for classes. If there's only one, it'll just equip. Otherwise, it lets you pick which outfit!
+	var/list/ctf_gear = list("Rifleman" = /datum/outfit/ctf, "Assaulter" = /datum/outfit/ctf/assault, "Marksman" = /datum/outfit/ctf/marksman)
+	var/list/instagib_gear = list("Instagib" = /datum/outfit/ctf/instagib)
+	var/list/default_gear
```

Comment:
```suggestion
	var/list/default_gear = list()
```

any reason why we don't type itlike this ?

Reply:
I don't think it matters but correct me if it does. The default gear var is overridden during init.

Reply:
it's a bit more """type""" safe to do it like this